cmd/kops-controller/pkg/server/server.go

```diff
@@ -140,7 +140,7 @@ func (s *Server) bootstrap(w http.ResponseWriter, r *http.Request) {
 
 	ctx := r.Context()
 
-	id, err := s.verifier.VerifyToken(ctx, r.Header.Get("Authorization"), body, s.opt.Server.UseInstanceIDForNodeName)
+	id, err := s.verifier.VerifyToken(ctx, r, r.Header.Get("Authorization"), body, s.opt.Server.UseInstanceIDForNodeName)
```

Comment:
Do we need to pass in both `r` the token here? Can't we fetch the header from `r` within the function?

Reply:
we can, but I tried to keep it simple for most of the providers. Instead parsing headers in each provider, I still like the way that authorization header is pre-parsed here. If they want to more advanced stuff the `r` is available (like openstack needs)